### PR TITLE
feat: group headers in pivot tables

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #5498 

### Description:

https://github.com/lightdash/lightdash/assets/9117144/93847f41-a440-41d2-bfb7-8699be4c8f66

### CODE REVIEW

**Comment by TuringLovesDeathMetal** [comment]
I think I broke it :( 

I'd expect this to have the statuses grouped together instead of split like they are now: ([example](https://lightdash-pr-5585.onrender.com/projects/36eb3bdf-0b74-4c66-974e-1a9745723113/tables/orders?create_saved_chart_version=%7B%22tableName%22%3A%22orders%22%2C%22metricQuery%22%3A%7B%22dimensions%22%3A%5B%22orders_order_date_week%22%2C%22orders_status%22%5D%2C%22metrics%22%3A%5B%22orders_average_order_size%22%2C%22orders_total_completed_order_amount%22%5D%2C%22filters%22%3A%7B%7D%2C%22sorts%22%3A%5B%7B%22fieldId%22%3A%22orders_order_date_week%22%2C%22descending%22%3Atrue%7D%2C%7B%22fieldId%22%3A%22orders_status%22%2C%22descending%22%3Afalse%7D%5D%2C%22limit%22%3A500%2C%22tableCalculations%22%3A%5B%5D%2C%22additionalMetrics%22%3A%5B%5D%7D%2C%22pivotConfig%22%3A%7B%22columns%22%3A%5B%22orders_status%22%2C%22orders_order_date_week%22%5D%7D%2C%22tableConfig%22%3A%7B%22columnOrder%22%3A%5B%22orders_order_date_week%22%2C%22orders_status%22%2C%22orders_average_order_size%22%2C%22orders_total_completed_order_amount%22%5D%7D%2C%22chartConfig%22%3A%7B%22type%22%3A%22table%22%2C%22config%22%3A%7B%22showColumnCalculation%22%3Afalse%2C%22showRowCalculation%22%3Afalse%2C%22showTableNames%22%3Atrue%2C%22columns%22%3A%7B%7D%2C%22hideRowNumbers%22%3Afalse%2C%22conditionalFormattings%22%3A%5B%5D%2C%22metricsAsRows%22%3Afalse%7D%7D%7D))

**Comment by TuringLovesDeathMetal** [comment]
^I think it's because I have it ordered by status second, but grouped by status first

**Comment by TuringLovesDeathMetal** [comment]
hmmm @ZeRego how hard is it to _**always**_ have the same values merge cells? (even if they aren't ordered first/second). 

It feels a bit unintuitive to have to order them in the results table a certain way to get them to merge in the table ciz.

**Comment by joaoviana** [comment]
@ZeRego 

I went to this example that @TuringLovesDeathMetal [shared](https://lightdash-pr-5585.onrender.com/projects/36eb3bdf-0b74-4c66-974e-1a9745723113/tables/orders?create_saved_chart_version=%7B%22tableName%22%3A%22orders%22%2C%22metricQuery%22%3A%7B%22dimensions%22%3A%5B%22orders_order_date_week%22%2C%22orders_status%22%5D%2C%22metrics%22%3A%5B%22orders_average_order_size%22%2C%22orders_total_completed_order_amount%22%5D%2C%22filters%22%3A%7B%7D%2C%22sorts%22%3A%5B%7B%22fieldId%22%3A%22orders_order_date_week%22%2C%22descending%22%3Atrue%7D%2C%7B%22fieldId%22%3A%22orders_status%22%2C%22descending%22%3Afalse%7D%5D%2C%22limit%22%3A500%2C%22tableCalculations%22%3A%5B%5D%2C%22additionalMetrics%22%3A%5B%5D%7D%2C%22pivotConfig%22%3A%7B%22columns%22%3A%5B%22orders_status%22%2C%22orders_order_date_week%22%5D%7D%2C%22tableConfig%22%3A%7B%22columnOrder%22%3A%5B%22orders_order_date_week%22%2C%22orders_status%22%2C%22orders_average_order_size%22%2C%22orders_total_completed_order_amount%22%5D%7D%2C%22chartConfig%22%3A%7B%22type%22%3A%22table%22%2C%22config%22%3A%7B%22showColumnCalculation%22%3Afalse%2C%22showRowCalculation%22%3Afalse%2C%22showTableNames%22%3Atrue%2C%22columns%22%3A%7B%7D%2C%22hideRowNumbers%22%3Afalse%2C%22conditionalFormattings%22%3A%5B%5D%2C%22metricsAsRows%22%3Afalse%7D%7D%7D)


And I noticed a duplicate `completed` on the status column. Is this an already existing bug or?

https://github.com/lightdash/lightdash/assets/7611706/f7544c63-ad9a-430b-aaaf-1aa8820ebacb

**Comment by ZeRego** [comment]
> hmmm @ZeRego how hard is it to always have the same values merge cells? (even if they aren't ordered first/second).
> It feels a bit unintuitive to have to order them in the results table a certain way to get them to merge in the table ciz.

@TuringLovesDeathMetal Changing this would require some work. As we discussed privately we are not making this change at the moment. 

> And I noticed a duplicate completed on the status column. Is this an already existing bug or?

@joaoviana Nice catch, there was a problem with a sorting function. It was sorting as if the values were strings instead of numbers. Fixed :)

File: packages/frontend/src/hooks/pivotTable/pivotQueryResults.ts [hunk 1]
```diff
@@ -149,7 +149,7 @@ const getColSpanByKey = (
     obj: RecursiveRecord<number>,
     keys: string[],
 ): number => {
-    const allIndices = getAllIndicesByKey(obj, keys).sort();
+    const allIndices = getAllIndicesByKey(obj, keys).sort((a, b) => a - b);
```
Comment (joaoviana) [inline packages/frontend/src/hooks/pivotTable/pivotQueryResults.ts:152]: 🙌

**Review by joaoviana — APPROVED** [review]

**Pull request merged**